README.md

```diff
@@ -1 +1,54 @@
-# designing_microservices_course
\ No newline at end of file
+# Designing microservices course
+This repository is intended for demonstration purposes for students during the "Basics of microservices architecture 
+course" which is going to be taught in 2019 autumn term for 4 years students if ITMO university, department of 
+informational systems.
+
+## Disclaimer
+Project is created to demonstrate main approaches and some of patterns which are used for creating microservices-based
+system. Some of architectural decisions were made especially for complicating system in order for us to have a room
+for applying some of patters, some were made to simplify system to order for us not having to implement real-world 
+business logic. Some of the services or business logic are just a mocks. We also didn't aim to create perfect code
+but tried to make it clear, tested, logged and  without duplications. But it still can't be considered as some best
+practices.
+
+## Overview
+
+### Business process description
+Project consists of several services communicating to ensure proper execution of some business processes. TODO sukhoa : 
+here should be description of the system.
+
+### Architecture overview
+Typically each individual service is just a small monolith or hexagonal application responsible for data and operations 
+from some certain business area. For example order management, user management etc. Service may have a number of inbound
+channels and outbound channels. Inbound channels enables service to get some data from the outside world. It may be 
+different types of data - client request, some event, resource state or whatever. Inbound channels in our system are 
+represented by REST API of the service and the queues or topics on which the service is subscribed. The REST API calls 
```

Comment:
~~The~~ REST API calls may sometimes overlay or simpl**y** duplicate ~~the~~ messaging API logic